bin/pulumilocal

```diff
@@ -124,22 +193,79 @@ def set_localstack_pulumi_config(args: argparse.Namespace):
-    parser.add_argument("-C", "--cwd", help="run in this directory",
+    parser.add_argument("-C", "--cwd", help="Run pulumi as if it had been started in another directory",
                         required=False,
                         type=str)
+    parser.add_argument("--config-file", help="Use the configuration values in the specified file rather than detecting the file name",
+                        required=False,
+                        type=str)
+    parser.add_argument("--non-interactive", help="Disable interactive mode for all commands",
+                        action="store_true",
+                        required=False)
     args, _ = parser.parse_known_args()
 
+    if NON_INTERACTIVE:
+        args.non_interactive = True
+
+    if not args.cwd:
+        args.cwd = "."
+
+    if os.environ.get("CONFIG_STRATEGY") not in CONFIG_STRATEGIES:
+        print(f"Config strategy: {os.environ.get('CONFIG_STRATEGY')}")
+        print(f'Config strategy is not recognised. Falling back to default ("{DEFAULT_CONFIG_STRATEGY}").')
+
     # If this is a pulumi deployment command, update the stack with LocalStack AWS config
-    if args.command in ["up", "destroy", "preview", "cancel"]:
+    if args.command in PROXIED_CMDS:
+        if not args.stack:
+            try:
+                args.stack = subprocess.run(
+                    [PULUMI_CMD, "stack", "--show-name", "--non-interactive", "-C", args.cwd],
+                    env=os.environ,
+                    check=True,
+                    capture_output=True
+                ).stdout.decode("utf-8").strip("\n")
+            except (subprocess.CalledProcessError) as e:
+                print(e.stderr.decode("utf-8"))
+                exit(1)
+        config_file_path = get_stack_config_file_path(args)
+        if not args.non_interactive:
+            check_stack_config_file(config_file_path)
+        if CONFIG_STRATEGY == "override":
+            override_config_file = os.path.join(os.path.dirname(config_file_path), f"Pulumi.{LOCAL_STACK_NAME}.yaml")
+            copyfile(src=config_file_path, dst=override_config_file)
+            args.config_file = override_config_file
+        elif CONFIG_STRATEGY == "separation":
+            print(f"Creating stack {LOCAL_STACK_NAME}, if not existing...")
+            subprocess.run(
+                [PULUMI_CMD, "stack", "init", "--non-interactive", "-s", LOCAL_STACK_NAME, "-C", args.cwd],
+                env=os.environ,
+                check=False,
+                stderr=subprocess.DEVNULL
+            )
+            override_config_file = os.path.join(os.path.dirname(config_file_path), f"Pulumi.{LOCAL_STACK_NAME}.yaml")
+            if override_config_file != config_file_path:
+                copyfile(src=config_file_path, dst=override_config_file)
+                args.config_file = override_config_file
         print("Updating this Stack with LocalStack config")
         set_localstack_pulumi_config(args)
```

Comment:
This set of conditions does not include the `overwrite` config strategy - is that deliberate? Is it implicitly the behaviour of lines 254-255? If so are these lines definitely compatible with being run regardless of the config strategy?

Reply:
`overwrite` doesn't need extra steps like using a different config file (`override`) or creating a new stack (`separate`), hence no need to include here, it can fall through to reconfigure the original stack config.